apps/web/src/pages/Profile/ProfilePage/ProfilePage.tsx

```diff
@@ -131,7 +131,18 @@ const ProfilePage: React.FunctionComponent = () => {
   const dataToUser = (input: GetMeQuery): User | undefined => {
     if (input) {
       if (input.me) {
-        return input.me;
+        return {
+          ...input.me,
+          poolCandidates: input.me.poolCandidates?.map((candidate) => {
+            if (candidate) {
+              return {
+                ...candidate,
+                pool: { id: candidate.poolAdvertisement?.id || "" },
```

Comment:
Ah, I see now why you might want to make PoolCandidate.pool nullable. 😅 While I still don't think we should change the schema at this time, this is a good cautionary tale of the downsides of making a type more restrictive.

I congratulate you on getting the Typescript magic to work out, but does ProfileForm ever _use_ pool candidates? It might just be easier to remove poolCandidates from the getMe operation in profileOperations.graphql.

Reply:
Good point! Changed with [529108b](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5927/commits/529108bb1075c579ea4b05b120f56796402d4492).